evennia/contrib/aws-s3-cdn.py

```diff
@@ -0,0 +1,847 @@
+"""
+This plugin migrates the Web-based portion of Evennia,
+namely images, javascript, and other items located
+inside staticfiles into Amazon AWS (S3) for hosting.
+
+INSTALLATION:
+
+1) If you don't have an AWS S3 account, you should create one now.
+
+Credentials required are an AWS IAM Access Key and Secret Keys,
+which can be generated/found in the AWS Console.
+
+Example IAM Control Policy Permissions, if desired:
+
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "evennia",
+            "Effect": "Allow",
+            "Action": [
+                "s3:PutObject",
+                "s3:GetObjectAcl",
+                "s3:GetObject",
+                "s3:ListBucket",
+                "s3:DeleteObject",
+                "s3:PutObjectAcl"
+            ],
+            "Resource": [
+                "arn:aws:s3:::YOUR_BUCKET_NAME/*",
+                "arn:aws:s3:::YOUR_BUCKET_NAME"
+            ]
+        }
+    ],
+    [
+      {
+         "Sid":"evennia",
+         "Effect":"Allow",
+         "Action":[
+            "s3:CreateBucket",
+         ],
+         "Resource":[
+            "arn:aws:s3:::*"
+         ]
+       }
+    ]
+}
+
+Advanced Users: The second IAM statement, CreateBucket, is only needed
+for initial installation. You can remove it later, or you can
+create the bucket and set the ACL yourself before you continue.
+
+2) This package requires the dependency "boto3," the official
+AWS python package. You can install it with 'pip install boto3'
+while inside your evennia virtual environment (or, simply
+in your shell if you don't use a virtual environment).
+
+3) Customize the variables defined below in secret_settings.py,
+then run 'evennia stop', 'evennia start', 'evennia collectstatic'
```

Comment:
`evennia collectstatic` should not be necessary; the `evennia start` will run this silently. Also, `evennia stop` followed by `evennia start` is the same as calling `evennia reboot` (which will work also if the server was not previously started).

Reply:
Noted, fixing.